backend/hitas/oracle_migration/runner.py

```diff
@@ -847,3 +876,218 @@ def create_codes(
         new.save()
 
     return retval
+
+
+def create_apartment_sales(connection: Connection, converted_data: ConvertedData) -> None:  # noqa: C901
+    def get_all_apartment_sales(apartment_oracle_id) -> list[hitas_monitoring]:
+        """Fetch all sales for given apartment"""
+        return list(
+            connection.execute(
+                select(hitas_monitoring)
+                .where(hitas_monitoring.c.apartment_id == apartment_oracle_id)
+                .order_by(asc(hitas_monitoring.c.purchase_date), asc(hitas_monitoring.c.id))
+            )
+        )
+
+    def create_sale_for_original_ownership() -> None:
+        """
+        Create a sale for an apartment that is never sold and still held by the original owner
+
+        Ownerships are already in the database, so bulk_create can't be used.
+        """
+        sale = ApartmentSale.objects.create(
+            apartment=v,
+            notification_date=v.first_purchase_date,
+            purchase_date=v.first_purchase_date,
+            purchase_price=v.purchase_price,
+            apartment_share_of_housing_company_loans=v.primary_loan_amount,
+            exclude_in_statistics=False,
+        )
+        # Link apartment ownerships to this sale
+        v.ownerships.all().update(sale=sale)
+
+    def get_or_create_buyers(sale) -> list[Owner]:
+        """Get buyers (Owner) from already converted data, or create new ones."""
+        buyers = []
+
+        # Handle both Buyer 1 and Buyer 2
+        for i in ["1", "2"]:
+            name = sale["buyer_name_" + i]
+
+            # Owner is unknown, can be skipped
+            if name == "Ei tiedossa":
+                continue
+
+            if name:
```

Comment:
Maybe invert this check (if not name: continue) to get rid of an indentation level?